amazon/ion/ion_py_objects.py

```diff
@@ -0,0 +1,605 @@
+from decimal import Decimal
+from collections.abc import MutableMapping
+
+from amazon.ion.core import IonType, IonEvent
+from amazon.ion.symbols import SymbolToken
+
+
+class IonPyNull_new(object):
```

Comment:
Okay, I will keep this comment unsolved until renaming them. 

I named them `object`, `object_new` and `object_original` so that we can separate them in the existing tests for easier debug and performance comparisons.

Reply:
Please confirm this is still the plan.

Reply:
Yep, I'll rename it once we address other comments. I'm using these class names for easier before/after performance comparison...

Reply:
Sorry I missed the updated question. Would you also refer to removing the old IonPyObject? yeah I think we can completely switch to the _new IonPyObject and rename it to the official `IonPyObject`. This won't cause breaking changes.

Reply:
The latest commit b6fe2df renames IonPyObject_new to IonPyObject, we can safely remove `IonPyObject_original` now.